File: src/io/fdeitylink/keroedit/util/Logger.java
package io.fdeitylink.keroedit.util;

import java.util.logging.LogRecord;
import java.util.logging.FileHandler;
import java.util.logging.Level;

import java.io.IOException;

//TODO: Specify log Levels?

public final class Logger {
    private Logger() {

    }

    public static void logMessage(final String message) {
        try {
            final FileHandler logFile = new FileHandler("error.log");
            logFile.publish(new LogRecord(Level.ALL, message));
            logFile.close();
        }
        catch (final IOException except) {
            System.err.println("Error: " + message);
        }
    }

    public static void logThrowable(final Throwable t) {
        logThrowable("", t);
    }

    public static void logThrowable(final String message, final Throwable t) {
        final StringBuilder sBuilder = new StringBuilder(message).append('\n');
        sBuilder.append(t.getClass().getName()).append(": ").append(t.getMessage());

        for (final StackTraceElement element : t.getStackTrace()) {
            sBuilder.append("\n\t").append(element);
        }
        final String finalMessage = sBuilder.toString();

        try {
            final FileHandler logFile = new FileHandler("error.log");
            logFile.publish(new LogRecord(Level.ALL, finalMessage));
            logFile.close();
        }
        catch (final IOException ex) {
            System.err.println(finalMessage);
        }
    }
}

File: src/io/fdeitylink/util/Logger.kt
/*
 * TODO:
 * Allow specifying Log Levels?
 * Remove @JvmOverloads annotation on logThrowable when everything is converted to Kotlin
 * Am I properly appending log records rather than overwriting?
 */
package io.fdeitylink.util

import java.util.logging.LogRecord
import java.util.logging.FileHandler
import java.util.logging.Level

import java.io.IOException

object Logger {
    fun logMessage(message: String, logFile: String = "error.log") {
        try {
            val handle = FileHandler(logFile)
            handle.publish(LogRecord(Level.ALL, message))
            handle.close()
        }
        catch (except: IOException) {
            System.err.println(message)
        }
    }

    @JvmOverloads
    fun logThrowable(message: String = "", t: Throwable, logFile: String = "error.log") {
        val builder = StringBuilder(message).append('\n')
        builder.append("${t.javaClass.name}: ${t.message}")

        val stackTrace = t.stackTrace
        for (element in stackTrace) {
            builder.append("\n\t$element")
        }

        val finalMessage = builder.toString()

        try {
            val handle = FileHandler(logFile)
            handle.publish(LogRecord(Level.ALL, finalMessage))
            handle.close()
        }
        catch (except: IOException) {
            System.err.println(finalMessage)
        }
    }
}